Orange/statistics/util.py

```diff
@@ -231,3 +231,70 @@ def weighted_mean():
             np.zeros(X.shape[1]),
             nans,
             X.shape[0] - nans))
+
+
+def _sparse_has_zeros(x):
+    """ Check if sparse matrix contains any implicit zeros. """
+    return np.prod(x.shape) != x.data.size
+
+
+def _nan_min_max(x, axis=0, func=None):
+    if not sp.issparse(x):
+        return func(x, axis=axis)
+    elif axis is None:
+        extreme = func(x.data, axis=axis) if x.data.size else float('nan')
+        if _sparse_has_zeros(x):
+            extreme = func([0, extreme])
+        return extreme
+    else:
+        r = []
+        if axis == 0:
+            x = x.T
+
+        # TODO check & transform to correct format
+
+        for row in x:
+            values = row.data
+            extreme = func(values) if values.size else float('nan')
+            if _sparse_has_zeros(row):
+                extreme = func([0, extreme])
+            r.append(extreme)
+        return np.array(r)
+
+
+def nanmin(x, axis=None):
+    """ Equivalent of np.nammin that supports sparse or dense matrices. """
+    return _nan_min_max(x, axis, np.nanmin)
+
+
+def nanmax(x, axis=None):
+    """ Equivalent of np.nammax that supports sparse or dense matrices. """
+    return _nan_min_max(x, axis, np.nanmax)
+
+
+def average(x):
+    """ Equivalent of np.average that supports sparse or dense matrices. """
+    if not sp.issparse(x):
+        return np.average(x)
+    else:
+        n_nans = np.sum(np.isnan(x.data))
+        if n_nans:
+            return float('nan')
+        else:
+            n_values = np.prod(x.shape) - n_nans
+            return np.nansum(x.data) / n_values
+
+
+def unique(x, return_counts=True):
+    """ Equivalent of np.unique that supports sparse or dense matrices. """
+    if not sp.issparse(x):
+        return np.unique(x, return_counts=return_counts)
+    else:
```

Comment:
`else` is unnecessary here and in other functions, which first check if x is not sparse and return something.
It just adds an extra indentation to all of the actual function body.